plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/actions/RemoveWatchExpressionAction.java

```diff
@@ -14,46 +14,45 @@
 
 import com.google.inject.Inject;
 import java.util.Collections;
+import org.eclipse.che.api.debug.shared.model.WatchExpression;
 import org.eclipse.che.ide.api.action.AbstractPerspectiveAction;
 import org.eclipse.che.ide.api.action.ActionEvent;
 import org.eclipse.che.plugin.debugger.ide.DebuggerLocalizationConstant;
 import org.eclipse.che.plugin.debugger.ide.DebuggerResources;
 import org.eclipse.che.plugin.debugger.ide.debug.DebuggerPresenter;
-import org.eclipse.che.plugin.debugger.ide.debug.changevalue.ChangeValuePresenter;
 
 /**
- * Action which allows change value of selected variable with debugger
+ * Action allows remove debugger watch expression from debugger tree.
  *
- * @author Mykola Morhun
+ * @author Oleksandr Andriienko
  */
-public class ChangeVariableValueAction extends AbstractPerspectiveAction {
+public class RemoveWatchExpressionAction extends AbstractPerspectiveAction {
 
-  private final ChangeValuePresenter changeValuePresenter;
   private final DebuggerPresenter debuggerPresenter;
+  private WatchExpression selectedExpression;
 
   @Inject
-  public ChangeVariableValueAction(
+  public RemoveWatchExpressionAction(
       DebuggerLocalizationConstant locale,
       DebuggerResources resources,
-      ChangeValuePresenter changeValuePresenter,
       DebuggerPresenter debuggerPresenter) {
     super(
         Collections.singletonList(PROJECT_PERSPECTIVE_ID),
-        locale.changeVariableValue(),
-        locale.changeVariableValueDescription(),
+        locale.removeWatchExpression(),
+        locale.removeWatchExpressionDescription(),
         null,
-        resources.changeVariableValue());
-    this.changeValuePresenter = changeValuePresenter;
+        resources.removeExpressionBtn());
     this.debuggerPresenter = debuggerPresenter;
   }
 
   @Override
-  public void actionPerformed(ActionEvent e) {
-    changeValuePresenter.showDialog();
+  public void actionPerformed(ActionEvent event) {
+    debuggerPresenter.onRemoveExpressionBtnClicked(selectedExpression);
```

Comment:
Don't store `selectedExperssion` as a global variable. Retrieve it inside the method.